### PR TITLE
Migrate away from RIO, add Blammo, use extracted API client

### PR DESCRIPTION
**Pull request by pbrisbin**
We were using `RIO` here for two reason:

- Its Prelude
- Logging

At the time, `RIO` was a contender to replace `Freckle.App` and solve
our custom Prelude needs. This has not materialized, and we've moved
further and further away from it. The current approach is
`Freckle.App.Prelude` and a new Structured Logging library, Blammo.

This commit migrates to Blammo, but not `freckle-app`. The reason I did
not bring in freckle-app is that this project has some non-standard
behavior w.r.t. to logging configuration, which I wanted to preserve for
now. Making a project-local Prelude worked well enough.

I also extracted the API client bits to the `asana-hs` repository and
released that as the `asana` package on Hackage. I decided the api
client was more publicly-valuable, so I gave it the name "asana" and
renamed this package to "asana-cli" to avoid conflict. I plan to rename
this repository to match, but I'm not married to this name in
particular, so please speak up if you have thoughts.

(Note that naming the api client repository "asana-hs" was _not_ to
avoid a conflict, but because that's our convention for libraries with
generic names that may exist in many languages.)

With Blammo, the logs while running the CLI tool now look like this:

And if we ever try to put this in a deployed service, `LOG_FORMAT=json`
is now available.

### CODE REVIEW

File: bug-reproduction/Main.hs [hunk 1]
```diff
@@ -52,22 +53,22 @@ main = do
...
-                  , " completed " <> displayShow sCompletedAt
-                  , " lacks reproduced information"
-                  ]
+                when (isNothing sReproduced)
+                  $ logWarn
+                  $ "Story lacks reproduced information"
+                  :# [ "story" .= sName
+                     , "assignee" .= nName assignee
+                     , "completedAt" .= sCompletedAt
+                     ]
```
Comment (z0isch) [inline bug-reproduction/Main.hs:62]: 🔥

File: library/Asana/App.hs [hunk 2]
```diff
@@ -41,64 +46,92 @@ import Options.Applicative
...
-  asanaApiAccessKeyL =
-    lens appApiAccessKey $ \app appApiAccessKey -> app { appApiAccessKey }
+instance HasAsanaAccessKey (AppWith ext) where
+  asanaAccessKeyL =
+    lens appAsanaAccessKey $ \app appAsanaAccessKey -> app { appAsanaAccessKey }
 
 data Perspective = Pessimistic | Optimistic
 
-type AppM ext = RIO (AppWith ext)
+type AppM ext a = ReaderT (AppWith ext) (LoggingT IO) a
```
Comment (z0isch) [inline library/Asana/App.hs:73]: Well that was easy...

**Review by z0isch — APPROVED** [review]
Very cool! Ad-hoc log messages are always a bit 😬 so I am very appreciative of this change.

I'm not sure I'm much help for a name outside of `asana-cli` maybe something like `asana-tools`?

**Comment by mjgpy3** [comment]
[Away Rio!](https://www.youtube.com/watch?v=ojkNEucG6mI)

**Pull request merged**